airflow/providers/amazon/aws/triggers/sqs.py

```diff
@@ -0,0 +1,170 @@
+        return (
+            self.__class__.__module__ + "." + self.__class__.__qualname__,
+            {
+                "sqs_queue": self.sqs_queue,
+                "aws_conn_id": self.aws_conn_id,
+                "max_messages": self.max_messages,
+                "num_batches": self.num_batches,
+                "wait_time_seconds": self.wait_time_seconds,
+                "visibility_timeout": self.visibility_timeout,
+                "message_filtering": self.message_filtering,
+                "delete_message_on_reception": self.delete_message_on_reception,
+                "message_filtering_match_values": self.message_filtering_match_values,
+                "message_filtering_config": self.message_filtering_config,
+                "waiter_delay": self.waiter_delay,
+            },
+        )
+
+    @property
+    def hook(self) -> SqsHook:
+        return SqsHook(aws_conn_id=self.aws_conn_id)
+
+    async def poll_sqs(self, client: BaseAwsConnection) -> Collection:
+        """
+        Asynchronously poll SQS queue to retrieve messages.
+
+        :param client: SQS connection
+        :return: A list of messages retrieved from SQS
+        """
+        self.log.info("SqsSensor checking for message on queue: %s", self.sqs_queue)
+
+        receive_message_kwargs = {
+            "QueueUrl": self.sqs_queue,
+            "MaxNumberOfMessages": self.max_messages,
+            "WaitTimeSeconds": self.wait_time_seconds,
+        }
+        if self.visibility_timeout is not None:
+            receive_message_kwargs["VisibilityTimeout"] = self.visibility_timeout
+
+        response = await client.receive_message(**receive_message_kwargs)
+        return response
+
+    async def poke(self, client: Any):
+        message_batch: list[Any] = []
+
+        for _ in range(self.num_batches):
+            self.log.info("starting call to poll sqs")
+            response = await self.poll_sqs(client=client)
+            messages = process_response(
+                response,
+                self.message_filtering,
+                self.message_filtering_match_values,
+                self.message_filtering_config,
+            )
+
+            if not messages:
+                continue
+
+            message_batch.extend(messages)
+
+            if self.delete_message_on_reception:
+                self.log.info("Deleting %d messages", len(messages))
+
+                entries = [
+                    {"Id": message["MessageId"], "ReceiptHandle": message["ReceiptHandle"]}
+                    for message in messages
+                ]
+                response = await client.delete_message_batch(QueueUrl=self.sqs_queue, Entries=entries)
+
+                if "Successful" not in response:
+                    raise AirflowException(
+                        "Delete SQS Messages failed " + str(response) + " for messages " + str(messages)
+                    )
+        if not message_batch:
+            return None
```

Comment:
nitpick: we might not need the `None` here but either way works

```suggestion
            return
```

Reply:
I realized I can just return `message_batch` which is cleaner than the original solution.

Reply:
Sounds great. 🙌